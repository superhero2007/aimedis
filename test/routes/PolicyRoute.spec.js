import React from 'react';
import Router from 'react-router-dom/MemoryRouter';
import { renderToString } from 'react-dom/server';
import PolicyRoute from 'routes/PolicyRoute';

describe('routes/PolicyRoute', () => {
  it('should render the Policy Page for unauthenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/policy']}>
        <PolicyRoute
          exact
          path="/policy"
          component={() => (<div>Policy</div>)}
          isAuthenticated={false}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });

  it('should render the Policy Page for authenticated access', () => {
    const render = renderToString(
      <Router initialEntries={['/policy']}>
        <PolicyRoute
          exact
          path="/policy"
          component={() => (<div>Policy</div>)}
          isAuthenticated={true}
        />
      </Router>
    );

    expect(render).toMatchSnapshot();
  });
});
