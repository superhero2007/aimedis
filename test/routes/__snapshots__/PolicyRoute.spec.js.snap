// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes/PolicyRoute should render the Policy Page for authenticated access 1`] = `"<div data-reactroot=\\"\\">Policy</div>"`;

exports[`routes/PolicyRoute should render the Policy Page for unauthenticated access 1`] = `"<div data-reactroot=\\"\\">Policy</div>"`;
